linux/xdg/src/main/java/com/github/weisj/darkmode/platform/linux/xdg/FreedesktopInterface.kt

```diff
@@ -0,0 +1,67 @@
+package com.github.weisj.darkmode.platform.linux.xdg
+
+import org.freedesktop.dbus.annotations.DBusInterfaceName
+import org.freedesktop.dbus.connections.impl.DBusConnection
+import org.freedesktop.dbus.interfaces.DBusInterface
+import org.freedesktop.dbus.interfaces.DBusSigHandler
+import org.freedesktop.dbus.messages.DBusSignal
+import org.freedesktop.dbus.types.UInt32
+import org.freedesktop.dbus.types.Variant
+
+enum class ThemeMode {
+    LIGHT, DARK, ERROR
+}
+
+@DBusInterfaceName("org.freedesktop.portal.Settings")
+interface FreedesktopInterface : DBusInterface {
+    companion object {
+        const val APPEARANCE_NAMESPACE = "org.freedesktop.appearance"
+        const val COLOR_SCHEME_KEY = "color-scheme"
+
+        private val connection = DBusConnection.getConnection(DBusConnection.DBusBusType.SESSION)
+        private val freedesktopInterface: FreedesktopInterface = connection.getRemoteObject(
+            "org.freedesktop.portal.Desktop",
+            "/org/freedesktop/portal/desktop",
+            FreedesktopInterface::class.java
+        )
```

Comment:
Actually thinking about this some more I'd prefer these variables to be members (same goes for the functions below), with `XdgThemeMonitorService` holding an instance of it. Otherwise we uneccessarily initialize everything in here even If it isn't used (which may also prevent the plugin being unloaded)

Reply:
So do I move those back to the theme monitor service, or should I create a new class holding the connection and functions? Because I can't make them members of the interface...

Reply:
Does it need to be an interface? If not just make it a class.